### ISSUE TITLE
Tests fail silently. No debugging info available

### ISSUE THREAD

**Issue by sheenarbw** [alteryx/locust-grasshopper#16]
I have a scenario.yaml file and a bunch of tests and I've been able to get the tests to run in the past, but when things aren't working then they are very hard to debug. There are situations where the tests all "pass" without actually executing. 

I have the following files:

scenarios.yaml

```
example_scenario_1:
  test_file_name: 'test_mini_frontend.py'
  grasshopper_args:
    users: 1
    spawn_rate: 1
    runtime: 200
  grasshopper_scenario_args:
    thresholds:
      get home page:
        type: get
        limit: 2000
  tags:
    - 'smoke'
```

test_mini_frontend.py
```
foooooooo  # yes, an intentional error

# actually valid code 
```

I try to run the tests like so:

```
pytest scenarios.yaml
```

I am expecting some kind of error message. But instead, I get:

```
================================== test session starts ==================================
platform linux -- Python 3.10.6, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/sheena/workspace/Tilde/saturation-tester/src, configfile: pytest.ini
plugins: locust-grasshopper-1.3.1
collecting ... 
---------------------------------- live log collection ----------------------------------
2023-05-15 12:50:25 [ WARNING] Since no --tags param was specified, ALL scenarios in /home/sheena/workspace/Tilde/saturation-tester/src/scenarios.yaml will be run! (__init__.py:495)
collected 1 item                                                                        

scenarios.yaml::example_scenario_1 PASSED                                         [100%]

=================================== 1 passed in 0.14s ===================================
```

# Ideal solution:

If there are errors in the tests then those errors should result in tracebacks so that they can be fixed. 

Errors should never pass silently.
Unless explicitly silenced.

**Comment by alteryx-sezell** [comment]
I think this might have to do with the way that launch individual scenarios in their own pytest context. Something isn't getting reported back correctly to first process that was doing the collecting.

**Comment by alteryx-sezell** [comment]
Yep, here is the problem.
Line 441 in `fixtures/__init__.py`

`        pytest.main(args)`

originally, launching pytest.main looked like this

```
 return_code = pytest.main(args)
assert_that(return_code).is_equal_to(pytest.ExitCode.OK)
```

this piece of code has moved locations a few times, the return code bit somehow got lost along the way.

Note that most failures (things that come from asserts) will show up not being passed through the return_code. The return_code comes into to play when the pytest.main process isn't able to load the test file at all for some reason - syntax error, import errors, etc. If you have other use cases where it's failing silently, please report those as separate issues.

**Comment by jmfiola** [comment]
@alteryx-sezell @sheenarbw Made a PR for this: https://github.com/alteryx/locust-grasshopper/pull/17/files

**Comment by jmfiola** [comment]
This has now been fixed in locust-grasshopper version `1.3.2`

**Comment by jmfiola** [comment]
@sheenarbw Thanks for raising this! Our first issue from the community is exciting.
